apps/xscreensaver/xscreensaver.star

```diff
@@ -0,0 +1,551 @@
+    ["xflame", "XFlame", "XFlame", "Pulsing fire. It can also take an arbitrary image and set it on fire too. Written by Carsten Haitzler and many others; 1999.", True, ["Default", "All"]],
+    ["xjack", "XJack", "XJack", "A novel by Jack Torrance.  Written by Jamie Zawinski; 1997.", True, ["All"]],
+    ["xlyap", "XLyap", "XLyap", "The Lyapunov exponent makes pretty fractal pictures. Written by Ron Record; 1997.", True, ["Default", "Soothing", "All"]],
+    ["xmatrix", "XMatrix", "XMatrix", "The \"digital rain\" effect, as seen on the computer monitors in \"The Matrix\". Written by Jamie Zawinski; 1999.", True, ["Default", "Nerdy", "All"]],
+    ["xrayswarm", "XRaySwarm", "XRaySwarm", "Worm-like swarms of particles with vapor trails. Written by Chris Leger; 2000.", True, ["Default", "Jarring", "All"]],
+    ["xspirograph", "XSpirograph", "XSpirograph", "Simulates that pen-in-nested-plastic-gears toy from your childhood. Written by Rohit Singh; 2000.", True, ["All"]],
+    ["zoom", "Zoom", "Zoom", "Fatbits! Zooms in on a part of an image and scrolls, distorting each pixel with its own lens. Written by James Macnicol; 2001.", True, ["Default", "All"]],
+]
+
+def main(config):
+    # If anything goes wrong, we can always show the first hack
+    hack = 0
+
+    # If we were sent a hack...
+    if config.get("hack"):
+        # ...by number on the command-line, always render it
+        hacks = []
+        for _ in HACKS:
+            hacks.append(([_[0], _[1], _[2]]))
+        hack = int(config.get("hack"))
+
+    elif config.get("hackname"):
+        # ...by name on the command-line, always render it
+        hacks = []
+        index = 0
+        for _ in HACKS:
+            hacks.append(([_[0], _[1], _[2]]))
+            if config.get("hackname") == _[0]:
+                hack = index
+            index += 1
+
+    else:
+        # If we weren't sent a hack, assemble the list of enabled hacks by group
+        hacks = []
+        for _ in HACKS:
+            if (TEST and _[0] == config.get("show", HACKS[0][0])) or \
+               (not TEST and (config.get("group", GROUP_DEFAULT) in _[5]) and config.bool("hack_" + _[0], _[4])):
+                hacks.append(([_[0], _[1], _[2]]))
+
+        # If there are none, show an error message and tell the user how to fix it
+        if len(hacks) == 0:
+            return render.Root(
+                child = render.WrappedText(
+                    content = "XScreenSaver: Nothing is enabled. Add some!",
+                    align = "left",
+                ),
+            )
+
+        else:
+            # If we're randomly picking a hack...
+            if config.bool("random", True):
+                # Pick one, but make sure it's not the same as the last one we ran
+                hack_last = int(cache.get("hack") or 0)
+                hack = random.number(0, len(hacks) - 1)
```

Comment:
```suggestion
                random.seed(time.now().unix // 15)
                hack = random.number(0, len(hacks) - 1)
```

Reply:
This will allow for graphics to be chosen at random, while making the choice consistent over 15s time windows. Doing so will help our backend to effectively cache the rendered graphics.